### ISSUE TITLE
Text column is being ready by Metabase as a number (thus can't select values to filter on) 

### ISSUE THREAD

**Issue by zz-hh-aa** [metabase/metabase#49828]
### Describe the bug

I was trying to filter by a text value in a table and column I have previously filtered on before. I can't seem to select a value. When I click on one, the 'Add filter' button becomes clickable, but the value I selected isn't highlighted, and a new one `NaN` shows up. 

The field is definitely of type text but Metabase seems to think it is a number. This is specific to this column; other text columns work just fine. ![Image](https://github.com/user-attachments/assets/f842da3c-048d-4f67-b5e7-ef5ea9fff4d2)
![Image](https://github.com/user-attachments/assets/069c76c3-7f2a-4c48-82d4-08f87eb1d6b9)
![Image](https://github.com/user-attachments/assets/31c1d471-9074-4cd6-a156-af10c721de44)

### To Reproduce

I cannot reproduce this bug on another dataset or column, whether one of mine or the Metabase sample database. 

### Expected behavior

I should be able to select a value to filter on. It should show as selected after I click on it.

### Logs

[metabase_log_20241111.txt](https://github.com/user-attachments/files/17700870/metabase_log_20241111.txt)

### Information about your Metabase installation

```JSON
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/130.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "run-mode": "prod",
    "plan-alias": "",
    "version": {
      "date": "2024-09-16",
      "tag": "v0.50.26",
      "hash": "5a65f46"
    },
    "settings": {
      "report-timezone": null
    },
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.19"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.7.3"
      }
    }
  },
  "system-info": {
    "file.encoding": "UTF-8",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.24+8",
    "java.vendor": "Eclipse Adoptium",
    "java.vendor.url": "https://adoptium.net/",
    "java.version": "11.0.24",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.24+8",
    "os.name": "Linux",
    "os.version": "5.10.226-214.880.amzn2.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  }
}
```

### Severity

Blocking partial usage

### Additional context

https://github.com/user-attachments/assets/95a51322-cb3e-40b4-be0a-9209a7200686

**Comment by Axxoul** [comment]
@zz-hh-aa I am having the same issue, how did you fix this?

**Comment by zz-hh-aa** [comment]
@Axxoul I believe that the 'Fields in this table' was showing this column correctly as Text (Category). We tried changing the type to Text (Title) something else and then back to Text (Category), and that fixed it. What we think happened was that this column used to be a numeric field and after we converted the view to a text column, Metabase didn't correctly pick up the change. Recategorising it and then changing it back properly set it to a text field, if I remember all correctly!
